src/api/schema/components.rs

```diff
@@ -8,7 +8,7 @@ use std::{
 };
 use tokio::stream::{Stream, StreamExt};
 
-const INVARIANT: &str =
+pub const INVARIANT: &str =
     "It is an invariant for the API to be active but not have COMPONENTS. Please report this.";
```

Comment:
Agreed, I'll make that clearer.

FWIW, the use of 'invariant' here is due to this being an internal error, rather than something that the user has done wrong. Technically, this error is only invoked if there's an issue acquiring a lock on the components, so the message is even vaguer. I'll clear that up.